main.sh

```diff
@@ -57,6 +57,11 @@ main_application_environment_variables=$(jq -r '.SecretString | fromjson | to_en
 main_export_regex="^([A-Z0-9_])+[=]+(.*)"
 for ENV_LINE in $main_application_environment_variables; do
     if [[ $ENV_LINE =~ ${main_export_regex} ]]; then
+        key="${ENV_LINE%%=*}"
+        if [ $key == "AWS_ACCESS_KEY_ID" ] || [ $key == "AWS_SECRET_ACCESS_KEY" ]; then
+            echo -e "\033[1mWARNING: the code need to be refactored to call AWS credentials using different names, because have this credentials in secrets manager conflicts with the IAM role of the CodeBuild project!\033[0m"
```

Comment:
Please improve the message to:
    Clearly state why these variables cause a conflict (i.e., they override the IAM role credentials injected by CodeBuild).
    Recommend how to resolve the issue (e.g., use alternative variable names like APP_AWS_ACCESS_KEY_ID if absolutely necessary).
    Avoid suggesting a full code refactor if that's not required.

Suggested message:

"WARNING: Detected use of AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY in environment variables. 
These override the IAM role provided by CodeBuild and may cause authentication issues. 
Consider renaming these variables (e.g., APP_AWS_ACCESS_KEY_ID and APP_AWS_SECRET_ACCESS_KEY) to avoid conflicts."

This kind of message is more helpful for whoever is running into the issue.

Reply:
Thank you for this comment!
I did an update to turn the message more helpful following your suggestion.